core/modules/loaders.rs

```diff
@@ -171,6 +172,62 @@ impl ModuleLoader for ExtModuleLoader {
   }
 }
 
+/// A loader that is used in `op_lazy_load_esm` to load and execute
+/// ES modules that were embedded in the binary using `lazy_loaded_esm`
+/// option in `extension!` macro.
+pub(crate) struct LazyEsmModuleLoader {
+  sources: Rc<RefCell<HashMap<&'static str, ExtensionFileSource>>>,
+}
+
+impl LazyEsmModuleLoader {
+  pub fn new(
+    sources: Rc<RefCell<HashMap<&'static str, ExtensionFileSource>>>,
+  ) -> Self {
+    LazyEsmModuleLoader { sources }
+  }
+}
+
+impl ModuleLoader for LazyEsmModuleLoader {
+  fn resolve(
+    &self,
+    specifier: &str,
+    referrer: &str,
+    _kind: ResolutionKind,
+  ) -> Result<ModuleSpecifier, Error> {
+    Ok(resolve_import(specifier, referrer)?)
+  }
+
+  fn load(
+    &self,
+    specifier: &ModuleSpecifier,
+    _maybe_referrer: Option<&ModuleSpecifier>,
+    _is_dyn_import: bool,
+  ) -> Pin<Box<ModuleSourceFuture>> {
+    let sources = self.sources.borrow();
+    let source = match sources.get(specifier.as_str()) {
+      Some(source) => source,
+      None => return futures::future::err(anyhow!("Specifier \"{}\" cannot be lazy-loaded as it was not included in the snapshot.", specifier)).boxed_local(),
```

Comment:
Are they actually snapshotted, or are they just made available?

Reply:
Ooops, seems like I didn't change the error message completely. It was meant to say `it was not inlucded in the binary`.